d2/src/main/java/com/linkedin/d2/balancer/strategies/relative/StateUpdater.java

```diff
@@ -380,7 +380,7 @@ private long getAvgClusterLatency(Set<TrackerClient> trackerClients, Map<Tracker
 
     return callCountSum + outstandingCallCountSum == 0
         ? 0
-        : Math.round((latencySum + outstandingLatencySum) / (double) (callCountSum + outstandingCallCountSum));
+        : (long) Math.ceil((latencySum + outstandingLatencySum) / (double) (callCountSum + outstandingCallCountSum));
```

Comment:
This is actually a subtle bug in relative load balancer strategy. When calculating the average cluster latency, if it's, say 0.4ms, it will be rounded to 0ms. Then when we calculate the health state of each host, every host will be marked as unhealthy, because 0ms * high_relative_factor is still 0ms no matter what the factor is. So we should round-up for the average cluster latency calculation